pkg/netflow/server.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020-present Datadog, Inc.
+
+package netflow
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/netflow/config"
+	"github.com/DataDog/datadog-agent/pkg/netflow/flowaggregator"
+)
+
+var serverInstance *Server
+
+// Server manages netflow listeners.
+type Server struct {
+	Addr          string
+	config        *config.NetflowConfig
+	listeners     []*netflowListener
+	demultiplexer aggregator.Demultiplexer
+	flowAgg       *flowaggregator.FlowAggregator
+}
+
+// NewNetflowServer configures and returns a running SNMP traps server.
+func NewNetflowServer(demultiplexer aggregator.Demultiplexer) (*Server, error) {
```

Comment:
Your new server is not dependent of the `Demultiplexer`, it only uses its `Sender` (none of the rest, neither its flush mechanics). I guess it would be a better design if you receive Senders here.

Reply:
Good idea. Changed here https://github.com/DataDog/datadog-agent/pull/10976/commits/d421fc305dc1746dd8ab168c60cec95415b54b88